### DISCUSSION TITLE
Has anyone tried Atari 8-bit or ST?

### DISCUSSION THREAD

**Discussion by Sowden** [darthcloud/BlueRetro#404] [Q&A] [answered]
Hey there guys.  Has anyone here tried attaching blueretro to Atari 8-bit (400, 800, 1200, etc) or a ST (520, 1040, ST/F/M, E, etc) computer?  Either internally or externally?  Looking at the Sega Genesis hookup, it uses the same connector.  I'm not sure however if the wiring is any different.  And I'm also not sure if the controller port sends out the 3.3v that the board needs to operate correctly.  Anyone here might know the answer to that?  Thanks guys.

**Comment by Sowden** [comment]
Alright, so I've been looking at some pinouts for the Atari 8 computer and the ST.  It seems like the configuration is similar to the Genesis pinouts.  There's some info about it here.

https://info-coach.fr/atari/hardware/interfaces.php#The_Mouse_Joystick_Connectors
https://forums.atariage.com/topic/225470-the-atari-joystick-pinout-common-ground

It looks like fire is on pin 6 on all of them (it's the C button on the Genesis controller).  I don't have a ESP32 board yet.  But it looks like a external cable made for the Genesis would work for both of these computers.  Has anyone tried this or able to try it?  It would be great to add the Atari computers to it's compatibility list.

**Comment by darthcloud** [comment]
You probably want to use parallel 1p or parallel 2p

https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-Cables-Build-Instructions#parallel-2p-5v-6-buttons-each-adapter-cable

For power its hard to tell without trying worst case use external supply

**Comment by Sowden** [comment]
Your probably right.  It would be more like the 2600 because the Atari controllers were universal throught the majority of its consoles/computers.  And it would be interesting if mouse support is implamented for the ST because it has been for other systems, SNES comes to mind.  It would start to open up to other computers like the Commadore, but I do know that the connector is the same but the wiering is different.  If no one else can test, I'll hook mine up and report back.

**Comment by Sowden** [comment, marked as answer]
Hey there guys.  Well I wanted to come back to this as I've been working on it for quite some time.  I can confirm that Blueretro works for both the Atari 800 XL and Atari ST computers.  I bought a Retro Rosseta to work with both of these, as this product needs its additional LevelShifter upgrade board, and both will need an external USB power supply to power the device.  I used a USB-C cable into my cell phone plug and it worked just fine.  Here is some info on each.

Atari 800XL: This works right out of the box.  As discussed above, both of these will use the parallel 2p firmware flash.  And this one works with really no effort at all.  I am using a N64 NSO controller, so I mapped the d-pad to the joystick, and it plays like I'm using an Atari joystick.  Both players work, I've tested it using a Atari joystick testing program.  Its great.

Atari ST: This one had me scratching my head for a long while.It took me a long time to figure this all out, but the ST needs the mouse to be plugged into controller port 1 and the joystick in controller port 2.  So you will need two controllers for this, one to control the mouse and the second for the controller.  In the future I am going to see about a bluetooth mouse to take care of this.  But just like with the 800XL, I assigned the dpad to my N64 joystick and it worked great.  Just be sure to keep those things in mind and it will work as it should.

So I wanted to confirm this so that it can be added to the compatibility list.  I'll tell you, I am very happy with how this all works.  Its breathing new life into my old computers.  Thanks @darthcloud for all of your hard work.
